### PR TITLE
add context when return errors during datapath initialization

### LINKED ISSUE #17925: Improve logging when failing to create a veth device
## Proposal / RFE

**Is your feature request related to a problem?**

The base Ubuntu distribution for Raspberry Pi's does not ship with the veth kernel module. Trying to install cilium results in it failing with only a "operation not supported" message.

Thread in Slack: https://cilium.slack.com/archives/C53TG4J4R/p1637010685276600

**Describe the solution you'd like**

1. Cilium should log the failure to create the veth device.
2. The system requirements documentation should make mention of the veth kernel module.

### PR DESCRIPTION
**Pull request by kerthcet**
Add some context information when return errors for failed device create

Signed-off-by: kerthcet <kerthcet@gmail.com>

Fixes: #17925

### CODE REVIEW

**Comment by kerthcet** [comment]
PTAL @pchaigno

**Review by qmonnet — CHANGES_REQUESTED** [review]
Hi and thanks for the contribution! Checkpatch [complains](https://github.com/cilium/cilium/runs/4326206298?check_suite_focus=true#step:4:11) about the commit subject being too long, we're trying to keep them under 75 characters for better readability. Would you mind fixing it please? The change looks good to me otherwise.

Side note: this only partially addresses #17925, as the following is not covered in your PR:
> 2. The system requirements documentation should make mention of the veth kernel module.

**Comment by pchaigno** [comment]
> Side note: this only partially addresses #17925, as the following is not covered in your PR:
> > The system requirements documentation should make mention of the veth kernel module.

I think it's fine to ignore that second item for now. I've created https://github.com/cilium/cilium/issues/18030 to track that and I'm planning to take care of it.

File: pkg/datapath/loader/base.go
Comment (pchaigno) [inline pkg/datapath/loader/base.go]:
```suggestion
		return fmt.Errorf("failed to setup base devices in mode %s: %w", mode, err)
```

**Review by pchaigno — COMMENTED** [review]
Thanks for the pull request!

File: pkg/datapath/loader/base.go
Comment (kerthcet) [reply pkg/datapath/loader/base.go]: done, thanks for your advices.

**Comment by kerthcet** [comment]
> Hi and thanks for the contribution! Checkpatch [complains](https://github.com/cilium/cilium/runs/4326206298?check_suite_focus=true#step:4:11) about the commit subject being too long, we're trying to keep them under 75 characters for better readability. Would you mind fixing it please? The change looks good to me otherwise.
> 
thanks for your guys' kindly advices. Hoping to contribute to cilium community continuously.

**Review by pchaigno — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]
Looks good, thanks a lot!

**Comment by qmonnet** [comment]
CI failures are flakes, documented as:
- k8s-1.21-kernel-5.4
    - https://github.com/cilium/cilium/issues/17010
- k8s-1.22-kernel-4.19
    - https://github.com/cilium/cilium/issues/18045 (new)
    - https://github.com/cilium/cilium/issues/18014

They all look unrelated to this PR.

We have the reviews from code owners so I'm merging this PR.

**Pull request merged**
